dn_util.go

```diff
@@ -0,0 +1,216 @@
+package ldap
+
+import (
+	enchex "encoding/hex"
+	"errors"
+	"strings"
+)
+
+// When true, uses strings.EqualFold to compare the values of an RDN.
+// This is usually needed as true for most values, set to false when
+// your DNs contain case sensitive values.
+var RDNCompareFold bool = true
+
+var ErrDNNotSubordinate = errors.New("Not a subordinate")
+
+// Returns the stringified version of a *DN, the RDN values are escaped
+func (dn *DN) String() string {
+	var rdns []string
+	for _, r := range dn.RDNs {
+		var tv []string
+		for _, av := range r.Attributes {
+			tv = append(tv, strings.ToLower(av.Type)+"="+EscapeValue(av.Value))
+		}
+		rdns = append(rdns, strings.Join(tv, "+"))
+	}
+	return strings.Join(rdns, ",")
+}
+
+func EscapeValue(value string) (escaped string) {
```

Comment:
Needs more specific naming to distinguish it from other escape funcs already in the package. Also needs godoc